### ISSUE TITLE
Consider un-amalgamating Arbiter

### ISSUE THREAD

**Issue by valgur** [PDAL/PDAL#4331]
The amalgamated `arbiter.hpp` exports a handful of extremely broadly named macros:
https://github.com/PDAL/PDAL/blob/c5cd71881c45e19b5144f9ebe9a713ccbda2de1d/vendor/arbiter/arbiter.hpp#L3280-L3311

These are only used internally in .cpp files in non-amalgamated Arbiter, but become publicly visible in PDAL through these installed headers:

- pdal/kernels/private/PointlessLas.hpp
- pdal/io/EsriReader.hpp
- pdal/io/private/stac/Utils.hpp
- pdal/io/private/connector/Connector.hpp

Also breaks the build when trying to build PDAL with a newer non-vendored Eigen 3.4.0:
```
eigen3d88c0279cc26/p/include/eigen3/Eigen/src/Core/util/Meta.h:503:31: error: macro "F" requires 3 arguments, but only 1 given
  503 | struct result_of<F(ArgTypes...)> {
      |                               ^
In file included from pdal825376639aac5/b/src/io/EsriReader.hpp:44,
                 from pdal825376639aac5/b/src/io/EsriReader.cpp:35:
pdal825376639aac5/b/src/vendor/arbiter/arbiter.hpp:3298: note: macro "F" defined here
 3298 | #define F(x,y,z) ((x & y) | (~x & z))
 ```

**Comment by hobu** [comment]
> breaks the build when trying to build PDAL with a newer non-vendored Eigen 3.4.0:

So Eigen gets to make macros named `F` but Arbiter can't? 😄 

We will clean up the Arbiter leakage in the public headers. As for the private headers, well, they're private 🤷.

**Comment by hobu** [comment]
https://github.com/PDAL/PDAL/pull/4334 removes the macros in Arbiter that were causing trouble.
https://github.com/PDAL/PDAL/pull/4335 updates PDAL to Eigen 3.4

Happy to merge a PR that optionally used an external Eigen (but used the vendored one by default).

**Comment by hobu** [comment]
Also happy to merge a PR that optionally used an external laz-perf (but used the vendored one by default).

**Comment by hobu** [comment]
Still need to fix the leak of arbiter in `pdal/io/EsriReader.hpp`

**Comment by valgur** [comment]
Fixed by #4334. Thank you!
